modules/javascript/partytown-web-worker/load.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Module Name: Partytown Web Worker
+ * Description: Run JavaScript in a separate Web Worker with the help of Partytown library.
+ * Experimental: Yes
+ *
+ * @since   1.0.0
+ * @package performance-lab
+ */
+
+/**
+ * PartyTown Configuration
+ *
+ * @since 1.0.0
+ * @see https://partytown.builder.io/configuration
+ * @return void
+ */
+function perflab_web_worker_partytown_configuration() {
+	$config = array(
+		'lib'     => str_replace( site_url(), '', plugin_dir_url( __FILE__ ) ) . 'assets/js/partytown/',
+		'forward' => array(),
+	);
+
+	/**
+	 * Add configuration for PartyTown.
+	 *
+	 * @since 1.0.0
+	 * @param array $config Configuration for PartyTown.
+	 * @return array
+	 */
+	$config = apply_filters( 'perflab_partytown_configuration', $config );
+
+	?>
+	<script>
+		window.partytown = <?php echo wp_json_encode( $config ); ?>;
+	</script>
+	<?php
+}
+add_action( 'wp_head', 'perflab_web_worker_partytown_configuration', 1 );
+
+/**
+ * Initialize PartyTown
+ *
+ * @since 1.0.0
+ * @return void
+ */
+function perflab_web_worker_partytown_init() {
+	wp_enqueue_script(
+		'partytown',
+		plugin_dir_url( __FILE__ ) . 'assets/js/partytown/partytown.js',
+		array(),
+		PERFLAB_VERSION,
+		false
+	);
+}
+add_action( 'wp_enqueue_scripts', 'perflab_web_worker_partytown_init', 1 );
+
+/**
+ * Get all scripts tags which have a `partytown` dependency.
+ *
+ * @since 1.0.0
+ * @return void
+ */
+function perflab_web_worker_partytown_worker_scripts() {
+	$partytown_handles = perflab_get_partytown_handles();
+
+	foreach ( $partytown_handles as $partytown_handle ) {
+		add_filter(
+			'script_loader_tag',
+			/**
+			 * Add type="text/partytown" to script tag.
+			 *
+			 * @since 1.0.0
+			 * @param string $tag Script tag.
+			 * @param string $handle Script handle.
+			 * @param string $src Script source.
+			 * @param string $partytown_handle Script handle which have `partytown` dependency.
+			 *
+			 * @return string $tag Script tag with type="text/partytown".
+			 */
+			function( $tag, $handle, $src ) use ( $partytown_handle ) {
+				if ( $handle === $partytown_handle ) {
+					$create_script_tag = sprintf(
+						'<script type="text/partytown" src="%1s" id="%2s"></script>',
+						$src,
+						$handle . '-js'
+					);
+					$tag               = $create_script_tag;
+				}
+				return $tag;
+			},
+			10,
+			3
+		);
+	}
+}
+add_action( 'wp_print_scripts', 'perflab_web_worker_partytown_worker_scripts' );
+
+/**
+ * Helper function to get all scripts tags which has `partytown` dependency.
```

Comment:
Rather than using a partytown dependency, can we look for scripts that have a `partytown-worker` strategy (and maybe no dependencies)? 

That way partytown could be enabled with:
`wp_script_add_data( $handle, 'strategy', 'partytown-worker' );`

Reply:
While it's feasible to execute this, my inclination is towards considering `partytown` to be included in the `$deps` array as a dependency. 

This preference stems from the fact that when a script loads as `script[type="text/partytown"]`, it necessitates the presence of `partytown`. Thus, referring to it as a direct dependency seems logical.

 Additionally, by adhering to the dependency approach, merely registering the `partytown` script suffices, as WordPress automatically enqueues it if another script depends on it else it will not be enqueued to the front-end.

Reply:
That makes sense, however I can also see the use case for the `wp_script_add_data` approach. For example, an optimization plugin could use this to specify it wants a specific script loaded with partytown. I'm not sure you could add a dependency dynamically like that (I have to look). 

This is fine as is for now, I will check if maybe we can make it work both ways.